pkg/controller/service/topology/topology_controller.go

```diff
@@ -0,0 +1,246 @@
+		CreateFunc: func(e event.CreateEvent) bool {
+			return false
+		},
+	}
+
+	// Watch for Deployment Update and Delete event
+	err = c.Watch(&source.Kind{Type: &appsv1.Deployment{}}, &handler.EnqueueRequestForObject{}, pred)
+	if err != nil {
+		return err
+	}
+
+	// Watch for Service Update and Delete event
+	err = c.Watch(&source.Kind{Type: &corev1.Service{}}, &handler.EnqueueRequestForObject{}, pred)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Reconcile handles events related to changes to the App Topology Service deployment.
+// This includes events from service/dc named "ServiceName" in the namespace "ServiceNameSpace"
+func (r *ReconcileService) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	// Check if deployment exist or not, create one if absent
+	namespace := request.Namespace
+	name := request.Name
+	dExist, err := r.coreClient.AppsV1().Deployments(namespace).Get(name, metav1.GetOptions{})
+	if err != nil {
+		_, err = r.coreClient.AppsV1().Deployments(namespace).Create(newDeploymentConfigForAppService(nil, name, namespace))
+		if err != nil {
+			log.Error(err, "Failed to redeploy deployment")
+			return reconcile.Result{}, err
+		}
+	}
+	if dExist != nil && dExist.Name == name {
+		log.Info("Deployment already exist with the name : %s", dExist.Name)
+	}
+
+	// Check if service exist or not, create one if absent
+	svcExist, err := r.coreClient.CoreV1().Services(namespace).Get(name, metav1.GetOptions{})
+	if svcExist != nil && svcExist.Name == name {
+		return reconcile.Result{}, err
+	}
+	if err != nil {
+		newSvc, err := newService(namespace, name, 8080)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		_, err = r.coreClient.CoreV1().Services(namespace).Create(newSvc)
+		if err != nil {
+			fmt.Println("Failed to redeploy dc")
+		}
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func newDeploymentConfigForAppService(containerPorts []corev1.ContainerPort, serviceName string, serviceNameSpace string) *appsv1.Deployment {
+	labels := getLabelsForServiceDeployments(ServiceName)
+	//annotations := resource.GetAnnotationsForCR(cp)
+	if containerPorts == nil {
+		containerPorts = []corev1.ContainerPort{{
+			ContainerPort: 8080,
+			Protocol:      corev1.ProtocolTCP,
+		}}
+	}
+	return &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ServiceName,
+			Namespace: ServicesNamespace,
+			Labels:    labels,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Strategy: appsv1.DeploymentStrategy{
+				Type: "Recreate",
+			},
+			Replicas: int32Ptr(1),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      ServiceName,
+					Namespace: ServicesNamespace,
+					Labels:    labels,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  ServiceName,
+							Image: "quay.io/redhat-developer/app-service:latest", // TODO(Akash): parameterize this
```

Comment:
[Note]: This path will be parameterized in another PR

Reply:
This is a big technical debt!

Reply:
How about creating a `DevConsoleConfig` CRD and keep the docker image as an attribute there?
See this Console CRD for example https://github.com/openshift/console-operator/blob/master/manifests/00-crd-operator-config.yaml